streams/src/test/java/org/apache/kafka/streams/integration/PurgeRepartitionTopicIntegrationTest.java

```diff
@@ -0,0 +1,223 @@
+    public static void createTopics() throws InterruptedException {
+        CLUSTER.createTopic(INPUT_TOPIC, 1, 1);
+    }
+
+    @Before
+    public void setup() {
+        // create admin client for verification
+        Properties adminConfig = new Properties();
+        adminConfig.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        adminClient = AdminClient.create(adminConfig);
+
+        Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, APPLICATION_ID);
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(APPLICATION_ID).getPath());
+        streamsConfiguration.put(StreamsConfig.topicPrefix(TopicConfig.SEGMENT_MS_CONFIG), purgeIntervalMs);
+        streamsConfiguration.put(StreamsConfig.topicPrefix(TopicConfig.SEGMENT_BYTES_CONFIG), purgeSegmentBytes);
+        streamsConfiguration.put(StreamsConfig.producerPrefix(ProducerConfig.BATCH_SIZE_CONFIG), purgeSegmentBytes / 2);    // we cannot allow batch size larger than segment size
+        streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, purgeIntervalMs);
+
+        StreamsBuilder builder = new StreamsBuilder();
+
+        builder.stream(INPUT_TOPIC)
+               .groupBy(MockKeyValueMapper.SelectKeyKeyValueMapper())
+               .count();
+
+        System.out.println(builder.build().describe());
+
+        kafkaStreams = new KafkaStreams(builder.build(), new StreamsConfig(streamsConfiguration), time);
+    }
+
+    @After
+    public void shutdown() throws IOException {
+        if (kafkaStreams != null) {
+            kafkaStreams.close(30, TimeUnit.SECONDS);
+        }
+    }
+
+
+    @Test
+    public void shouldRestoreState() throws InterruptedException {
+        // produce some data to input topic
+        final Properties producerConfig = new Properties();
+        producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+
+        try (final KafkaProducer<Integer, Integer> producer =
+                     new KafkaProducer<>(producerConfig, new IntegerSerializer(), new IntegerSerializer())) {
+
+            for (int i = 0; i < 1000; i++) {
+                producer.send(new ProducerRecord<>(INPUT_TOPIC, i, i));
+            }
+        }
+
+        kafkaStreams.start();
+
+        TestUtils.waitForCondition(new RepartitionTopicCreated(), 60000,
+                "Repartition topic " + REPARTITION_TOPIC + " not created with the expected configs after 60000 ms.");
+
+        TestUtils.waitForCondition(
+                new RepartitionTopicPurged(new TopicSizeVerifier() {
```

Comment:
If I understand this correctly, we actually test if the topic is not purged yet? Thus, the name `RepartitionTopicPurged` is a little miss leading/confusing.

Reply:
Ack. I'm renaming to `RepartitionTopicVerified`, which is used for both `verify filled` and `verify purged`.